### PR TITLE
fix(popover): autoFocus without focusLock

### PR DESCRIPTION
**Pull request by lhbrennan**
#### Description

Previously, autoFocus did not work unless focusLock was true. Now, autoFocus and focusLock function correctly, independently of one another.

#### Scope
Patch: Bug Fix

### CODE REVIEW

File: src/app-nav-bar/user-menu.js [hunk 1]
```diff
@@ -120,6 +120,7 @@ export default function UserMenuComponent(props: {|
           {...userMenuProps}
         />
       )}
+      autoFocus={false}
```
Comment (lhbrennan) [inline src/app-nav-bar/user-menu.js:123]: There are multiple instances throughout the repo where we've used `Popover` in ways that assume autoFocus is false, when really it is true by default but just wasn't working correctly. In these cases we need to explicitly pass in `autoFocus={false}` to preserve existing functionality.

File: src/popover/__tests__/popover-focus-loop.scenario.js [hunk 2]
```diff
@@ -32,7 +32,7 @@ export function Scenario() {
         </div>
       )}
     >
-      <button>click</button>
```
Comment (lhbrennan) [inline src/popover/__tests__/popover-focus-loop.scenario.js:35]: The trigger for this popover is set to "hover", so this button should be named to reflect that.

**Comment by williamernest** [comment]
The failed tests indicate a time issue with the tests or the component. Can you investigate?

File: src/popover/__tests__/popover-autofocus.e2e.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    selectedEl,
+  );
+  activeEl.dispose();
+  return equal;
+};
+
+const inputSelector = 'input';
+
+describe('popover', () => {
+  it('contents are auto-focused but not focus-locked', async () => {
```
Comment (chasestarr) [inline src/popover/__tests__/popover-autofocus.e2e.js:28]: Do you mind adding a test for the new scenario file asserting that when the popover opens, the input selector is focused and after tab key press, the input selector is still focused?

**Review by chasestarr — APPROVED** [review]
Noting that this does change the default behavior of popover to auto-focus the first focusable element. This was always intended to be the default behavior, and I believe it was the default behavior until a regression was introduced. Given the historical context, this seems fine to merge now as a bug fix rather than waiting for a major release sometime early next year

**Pull request merged**